upstairs/src/backpressure.rs

```diff
@@ -88,40 +100,120 @@ impl Default for BackpressureConfig {
 #[derive(Copy, Clone, Debug)]
 pub struct BackpressureChannelConfig {
     /// When should backpressure start
-    pub start: u64,
-    /// Value at which backpressure goes to infinity
-    pub max: u64,
-    /// Scale of backpressure
-    pub scale: Duration,
+    pub start_value: u64,
+    /// Value at which backpressure is saturated
+    pub max_value: u64,
+
+    /// Characteristic scale of backpressure
+    ///
+    /// This scale sets the backpressure delay halfway between `start`_value and
+    /// `max_value`
+    pub delay_scale: Duration,
+
+    /// Maximum delay (returned at `max_value`)
+    pub delay_max: Duration,
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum BackpressureAmount {
+    Duration(Duration),
+    Saturated,
+}
+
+impl std::cmp::Ord for BackpressureAmount {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        match (self, other) {
+            (BackpressureAmount::Saturated, _) => std::cmp::Ordering::Greater,
+            (_, BackpressureAmount::Saturated) => std::cmp::Ordering::Less,
```

Comment:
I think there should be a third case before these two that returns Equal if both are saturated. otherwise our Ord/PartialOrd impl disagrees with our derived Eq/PartialEq

> The methods of this trait must be consistent with each other and with those of [PartialEq](https://doc.rust-lang.org/std/cmp/trait.PartialEq.html). The following conditions must hold:
> 1. a == b if and only if partial_cmp(a, b) == Some(Equal).

(from https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html)

Reply:
Good catch, done in dc3130f1bcea